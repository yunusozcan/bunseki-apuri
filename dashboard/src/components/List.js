import React from 'react';
import {Link} from "react-router-dom";
import ListItem from './ListItem';
import '../stylesheets/components/List.css';
import Button from "./Button";
import back from "../images/dashboard.svg";

function List(props) {
    const {list} = props;
    return (
        <React.Fragment>
            {list.map(listItem => (
                <ListItem key={listItem.id}>
                    <span className={"App-list-item-span"}>{listItem.webAdress}</span>
                    <Link to={"/dashboard/" + listItem.id}>
                        <Button>
                            <img src={back} className="App-dashboard-image" alt="dashboard"/>
                            <span>Dashboard</span>
                        </Button>
                    </Link>
                </ListItem>
            ))}
        </React.Fragment>
    );
}

export default List;
