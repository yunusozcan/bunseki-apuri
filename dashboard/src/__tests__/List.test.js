import React from 'react';
import List from '../components/List';
import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from "react-router-dom";

it('renders correctly', () => {
    const tree = renderer
        .create(
            <Router>
                <List list={[{id:1, webAdress: 'Test'}]}/>
            </Router>
            )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
