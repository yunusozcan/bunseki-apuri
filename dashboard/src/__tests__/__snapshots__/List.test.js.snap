// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App-list-item"
>
  <span
    className="App-list-item-span"
  >
    Test
  </span>
  <a
    href="/dashboard/1"
    onClick={[Function]}
  >
    <button
      className="App-button"
      type="button"
    >
      <img
        alt="dashboard"
        className="App-dashboard-image"
        src="dashboard.svg"
      />
      <span>
        Dashboard
      </span>
    </button>
  </a>
</div>
`;
